src/v/v8_engine/v8_batch_consumer.h

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright 2021 Vectorized, Inc.
+ *
+ * Licensed as a Redpanda Enterprise file under the Redpanda Community
+ * License (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * https://github.com/vectorizedio/redpanda/blob/master/licenses/rcl.md
+ */
+
+#pragma once
+
+#include "kafka/protocol/batch_consumer.h"
+#include "model/record.h"
+#include "v8_engine/executor.h"
+#include "v8_engine/scripts_dispatcher.h"
+
+namespace v8_engine {
+
+class v8_batch_consumer {
+public:
+    v8_batch_consumer(
+      scripts_dispatcher<executor_wrapper>& scripts_dispatcher, model::ntp ntp)
+      : _scripts_dispatcher(scripts_dispatcher)
+      , _ntp(std::move(ntp)) {}
+
+    v8_batch_consumer(const v8_batch_consumer& o) = delete;
+    v8_batch_consumer& operator=(const v8_batch_consumer& o) = delete;
+    v8_batch_consumer& operator=(v8_batch_consumer&& o) = delete;
+    v8_batch_consumer(v8_batch_consumer&& o) = default;
+    ~v8_batch_consumer() = default;
+
+    auto operator()(model::record_batch batch) {
+        // execute _script_dispatcher->run(ntp, batch);
+        return _kafka_serializer(std::move(batch));
+    }
+
+    auto end_of_stream() {
```

Comment:
this should return a `model::record_batch_reader` and you should have a `model::record_batch_reader::data_t` as a data member , pushing back transformed `model::record_batches` into this collection every iteration of `operator()` 

```
ss::stop_iteration operator()(model::record_batch&& rb) {
  try {
    _transformed_batches.push_back(_scripts_dispatcher.run(std::move(rb)));
  catch(...some exception...) {
    // log message
    co_return ss::stop_iteration::yes;
  }
  co_return ss::stop_iteration::no;
}

model::record_batch_reader end_of_stream() {
  return model::make_memory_record_batch_reader(std::move(_transformed_batches));
}
```

Reply:
There might even be a utility that follows this pattern in `model/record_batch_reader.h`

Reply:
I used `kafka_batch_serializer` like template for my consumer

Reply:
done
I' going to add integration with v8_runtime today